train.py

```diff
@@ -11,160 +11,16 @@
-    )
-    all_params = (
-        p
-        for param_group in optimizer.param_groups
-        for p in param_group["params"]
-    )
-
-    if config.checkpoint_path is not None:
-        logger.info(f"Loading checkpoint from {config.checkpoint_path}")
-
-        state_dict = cstorch.load(config.checkpoint_path)
-
-        model.load_state_dict(state_dict["model"])
-        if "optimizer" in state_dict:
-            optimizer.load_state_dict(state_dict["optimizer"])
-        if "lr_scheduler" in state_dict:
-            lr_scheduler.load_state_dict(state_dict["lr_scheduler"])
-        global_step = state_dict.get("global_step", 0)
-    else:
-        global_step = 0
-
-    @cstorch.checkpoint_closure
-    def save_checkpoint(step):
-        checkpoint_path = out_dir.joinpath(f"checkpoint_{step}.mdl")
-        state_dict = {
-            "model": model.state_dict(),
-            "optimizer": optimizer.state_dict(),
-            "lr_scheduler": lr_scheduler.state_dict(),
-            "global_step": step,
-            "model_config": asdict(model_config),
-        }
-        cstorch.save(state_dict, checkpoint_path)
-        logger.info(f"Saved checkpoint to {checkpoint_path}")
-
-    @cstorch.trace
-    def training_step(batch):
-        input_ids, labels = batch
-        loss = compiled_model(input_ids, labels)
-        loss.backward()
-        torch.nn.utils.clip_grad_norm_(list(all_params), config.max_gradient_norm)
-        optimizer.step()
-        lr_scheduler.step()
-        optimizer.zero_grad()
-        return loss
-
-    writer = cstorch.utils.tensorboard.SummaryWriter(
-        log_dir=out_dir.joinpath("train")
-    )
-
-    @cstorch.step_closure
-    def log_loss(loss, step):
-        rate = executor.profiler.rate()
-        global_rate = executor.profiler.global_rate()
-
-        logger.info(
-            f"| Step={step}, "
-            f"Loss={loss.item():.5f}, "
-            f"Rate={rate:.2f} samples/sec, "
-            f"GlobalRate={global_rate:.2f} samples/sec"
-        )
-        writer.add_scalar("loss", loss.item(), step)
-        writer.add_scalar("samples_per_second", global_rate, step)
-
-    data_path = os.path.join(config.dataset, "train.bin")
-    dataloader = cstorch.utils.data.DataLoader(
-        get_dataloader,
-        data_path,
-        config.sequence_length,
-        config.batch_size,
-        config.seed,
-    )
-    executor = cstorch.utils.data.DataExecutor(
-        dataloader,
-        num_steps=config.num_steps - global_step,
-        checkpoint_steps=config.checkpoint_steps,
-        cs_config=cs_config,
-        writer=writer,
-    )
-
-    for step, batch in enumerate(executor, start=global_step + 1):
-        if step > config.num_steps:
-            break
-        loss = training_step(batch)
-        log_loss(loss, step)
-        save_checkpoint(step)
-
-    logger.info("Training completed successfully!")
+from cerebras.modelzoo.common.utils.run.cli_pytorch import get_params_from_args
```

Comment:
I don't disagree with this change for getting it to work but I don't think we can continue to claim ~600 lines when we make library calls which hides all the code.

Reply:
Sorry, could you clarify on what you mean by hiding the code?

Reply:
Sorry for the late reply. The motivation for maintaining gigaGPT is to demonstrate easy model size scaling on Cerebras Hardware using only simple and readable pytorch code. 

If we delegate all the train.py code to a helper function, we can no longer claim that. One could argue that we scale only by hiding the complexity in a helper function as it is immediately not visible what happens behind `cerebras.modelzoo.common.run_utils.main`.

Reply:
ok, got it.